lib/metadataTypes/MetadataType.js

```diff
@@ -624,12 +624,15 @@ class MetadataType {
      *
      * @param {TYPE.MetadataTypeItem} metadataEntry a single metadata Entry
      * @param {string} uri rest endpoint for PATCH
+     * @param {boolean} [usePut] some update requests require PUT instead of PATCH
      * @returns {Promise} Promise
      */
-    static async updateREST(metadataEntry, uri) {
+    static async updateREST(metadataEntry, uri, usePut) {
```

Comment:
from memory put is needed for a new version, whereas patch is used to update an existing draft -> please confirm behaviour on this first :)

Reply:
for journey updates, the docs state PUT is what you need. In my tests, patch didn't work at all while put did what the docs promised.
if you want to create a draft for an already running journey (and there can be only one draft), then you need to run the create (POST) call. if, however, there is a draft and you run PATCH, you get errors, if you run PUT for that draft, it updates it. rather strange but working

https://developer.salesforce.com/docs/marketing/marketing-cloud/guide/putUpdateInteraction.html